middleware/src/rpcserver/mocks/Middleware.go

```diff
@@ -0,0 +1,347 @@
+// Code generated by mockery v1.0.0. DO NOT EDIT.
```

Comment:
Like that. How do we assure that the mocks are up-to-date?

Reply:
You need to re-run mockery. The test compilation will fail if they do not implement the full interface.

Reply:
Run `mockery -name Middleware` in the rpcserver directory. There are go:generate instructions for this in the rpcserver_test file. I also added them to the rpcserver file. In case your ide/editor is not configured to rerun `go generate` before every compilation run, you can execute the mockery command manually.

Reply:
Shouldn't that be included in the Makefile? Not good to be dependent on  IDE/editor config when (newly starting) developing the Middleware.

Reply:
Yeah, the more tooling we can hide in the Makefile, the better imo. I added it as a separate target.